pybamm/parameters/lithium_ion_parameters.py

```diff
@@ -224,6 +224,76 @@ def _set_dimensional_parameters(self):
             "Initial concentration in electrolyte [mol.m-3]"
         )
 
+        # mechanical prapmeters
+        if self.options["particle cracking"] is not None:
```

Comment:
you can remove these options. the mechanical parameter values only need to be present in `inputs/parameters/...` if the model uses a cracking model, so you can define the `Parameter` object here without causing errors

what was the error you were getting before?

Reply:
If without those conditions, there would be errors as below
```

======================================================================
ERROR: test_edit_param (test_parameters.test_parameters_cli.TestParametersCLI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/aishine/GEM/PyBaMM/tests/unit/test_parameters/test_parameters_cli.py", line 96, in test_edit_param
    with open(copied_path_parameters_file, "r") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpib26gnja/lithium-ion/anodes/tmpp7l8rlpm/parameters.csv'

======================================================================
FAIL: test_add_rm_param (test_parameters.test_parameters_cli.TestParametersCLI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/aishine/GEM/PyBaMM/tests/unit/test_parameters/test_parameters_cli.py", line 55, in test_add_rm_param
    self.assertTrue(os.path.isfile(new_parameter_filename))
AssertionError: False is not true
```

Reply:
hmm I'll take another look. I cloned you repo and was able to remove this line and not get this error. would be good to figure out what is causing it

Reply:
After updating from the pybamm develop branch, those errors are solved. Now I have removed those `if` conditions.